src/bin/pg_autoctl/monitor.c

```diff
@@ -1510,79 +1514,108 @@ printCurrentState(void *ctx, PGresult *result)
 		context->parsedOK = false;
 		return;
 	}
 
+	/* print size times a "-" charatter in the separator string */
 	for (int i = 0; i < maxNameSize; i++)
 	{
 		nameSeparatorHeader[i] = '-';
 	}
+	for (int i = 0; i < maxHostSize; i++)
+	{
+		hostSeparatorHeader[i] = '-';
+	}
 
 	if (pg_autoctl_debug)
 	{
-		fformat(stdout, "%*s | %6s | %5s | %17s | %17s | %8s | %6s\n",
-				maxNameSize, "Name",
-				"Group", "Node", "Current State", "Assigned State",
+		fformat(stdout, "%*s | %*s | %6s | %17s | %17s | %8s | %6s\n",
+				maxNameSize, "Name", maxHostSize, "Host:Port",
+				"Node", "Current State", "Assigned State",
 				"Priority", "Quorum");
 
-		fformat(stdout, "%*s-+-%6s-+-%5s-+-%17s-+-%17s-+-%8s-+-%6s\n",
+		fformat(stdout, "%*s-+-%*s-+-%6s-+-%17s-+-%17s-+-%8s-+-%6s\n",
 				maxNameSize, nameSeparatorHeader,
-				"------", "-----", "-----------------", "-----------------",
+				maxHostSize, hostSeparatorHeader,
+				"------", "-----------------", "-----------------",
 				"--------", "------");
 	}
 	else
 	{
-		fformat(stdout, "%*s | %6s | %5s | %17s | %17s\n",
+		fformat(stdout, "%*s | %*s | %5s | %17s | %17s\n",
 				maxNameSize, "Name",
-				"Group", "Node", "Current State", "Assigned State");
+				maxHostSize, "Host:Port",
+				"Node", "Current State", "Assigned State");
 
-		fformat(stdout, "%*s-+-%6s-+-%5s-+-%17s-+-%17s\n",
+		fformat(stdout, "%*s-+-%*s-+-%6s-+-%17s-+-%17s\n",
 				maxNameSize, nameSeparatorHeader,
-				"------", "-----", "-----------------", "-----------------");
+				maxHostSize, hostSeparatorHeader,
+				"------", "-----------------", "-----------------");
 	}
+
 	free(nameSeparatorHeader);
+	free(hostSeparatorHeader);
 
 	for (currentTupleIndex = 0; currentTupleIndex < nTuples; currentTupleIndex++)
 	{
 		char *name = PQgetvalue(result, currentTupleIndex, 0);
-
-		/* char *hostname = PQgetvalue(result, currentTupleIndex, 1); */
-		/* char *nodeport = PQgetvalue(result, currentTupleIndex, 2); */
+		char *host = PQgetvalue(result, currentTupleIndex, 1);
+		char *port = PQgetvalue(result, currentTupleIndex, 2);
 		char *groupId = PQgetvalue(result, currentTupleIndex, 3);
 		char *nodeId = PQgetvalue(result, currentTupleIndex, 4);
 		char *currentState = PQgetvalue(result, currentTupleIndex, 5);
 		char *goalState = PQgetvalue(result, currentTupleIndex, 6);
 		char *candidatePriority = PQgetvalue(result, currentTupleIndex, 7);
 		char *replicationQuorum = PQgetvalue(result, currentTupleIndex, 8);
 
+		char hostport[BUFSIZE] = { 0 };
+		char composedId[BUFSIZE] = { 0 };
+
+		sformat(hostport, sizeof(hostport), "%s:%s", host, port);
+		sformat(composedId, sizeof(hostport), "%s/%s", groupId, nodeId);
+
 		if (pg_autoctl_debug)
 		{
-			fformat(stdout, "%*s | %6s | %5s | %17s | %17s | %8s | %6s\n",
-					maxNameSize, name, groupId, nodeId,
+			fformat(stdout, "%*s | %*s | %6s | %17s | %17s | %8s | %6s\n",
+					maxNameSize, name,
+					maxHostSize, hostport,
+					composedId,
```

Comment:
I believe `composedId` should also have a `maxComposedIdSize`.

Right now the new state table is not lined up correctly in all cases.

With a small nodeid:
```
  Name |  Host:Port |  Node |     Current State |    Assigned State
-------+------------+--------+-------------------+------------------
node_1 |     l:5501 |    0/1 |            single |            single
```
And with a big nodeid:
```
       Name |          Host:Port |  Node |     Current State |    Assigned State
------------+--------------------+--------+-------------------+------------------
     node_1 |             l:5501 |    0/1 |           primary |           primary
node_100001 | 172.18.199.43:5502 | 0/100001 |         secondary |         secondary

``

Reply:
Fair enough, even though I think it's pushing it to edge cases, I don't see a good reason why we wouldn't expand our efforts in that direction too. Done now.